tests/test_config.py

```diff
@@ -247,6 +248,53 @@ async def mock_adr_0007_integrations_with_docs(
     return integrations
 
 
+@pytest.fixture
+async def mock_custom_validator_integrations(hass: HomeAssistant) -> list[Integration]:
+    """Mock integrations with custom validator."""
+    integrations = []
+
+    for domain in ("custom_validator_ok_1", "custom_validator_ok_2"):
+
+        def gen_async_validate_config(domain):
+            schema = vol.Schema(
+                {
+                    domain: vol.Schema(
+                        {
+                            vol.Required("host"): str,
+                            vol.Required("port", default=8080): int,
```

Comment:
It's a bit weird to mix required and default. If there's a default we generally make it optional since that's effectively what it is.

Reply:
Agreed, fixed